scipy/ndimage/_multimethods.py

```diff
@@ -0,0 +1,1085 @@
+import functools
+import numpy as np
+from scipy._lib.uarray import Dispatchable, all_of_type, create_multimethod
+from scipy.ndimage import _api
+from scipy.ndimage._backend import ndimage_index, ndimage_output
+
+
+__all__ = ['affine_transform', 'binary_closing', 'binary_dilation',
+           'binary_erosion', 'binary_fill_holes', 'binary_hit_or_miss',
+           'binary_opening', 'binary_propagation', 'black_tophat',
+           'center_of_mass', 'convolve', 'convolve1d', 'correlate',
+           'correlate1d', 'distance_transform_bf', 'distance_transform_cdt',
+           'distance_transform_edt', 'extrema', 'find_objects',
+           'fourier_ellipsoid', 'fourier_gaussian', 'fourier_shift',
+           'fourier_uniform', 'gaussian_filter', 'gaussian_filter1d',
+           'gaussian_gradient_magnitude', 'gaussian_laplace',
+           'generate_binary_structure', 'generic_filter', 'generic_filter1d',
+           'generic_gradient_magnitude', 'generic_laplace',
+           'geometric_transform', 'grey_closing', 'grey_dilation',
+           'grey_erosion', 'grey_opening', 'histogram', 'iterate_structure',
+           'label', 'labeled_comprehension', 'laplace', 'map_coordinates',
+           'maximum', 'maximum_filter', 'maximum_filter1d', 'maximum_position',
+           'mean', 'median', 'median_filter', 'minimum', 'minimum_filter',
+           'minimum_filter1d', 'minimum_position', 'morphological_gradient',
+           'morphological_laplace', 'percentile_filter', 'prewitt',
+           'rank_filter', 'rotate', 'shift', 'sobel', 'spline_filter',
+           'spline_filter1d', 'standard_deviation', 'sum', 'sum_labels',
+           'uniform_filter', 'uniform_filter1d', 'variance', 'watershed_ift',
+           'white_tophat', 'zoom']
+
+
+_mark_non_coercible = functools.partial(Dispatchable,
+                                        dispatch_type=ndimage_output,
+                                        coercible=False)
+_mark_index = functools.partial(Dispatchable,
+                                dispatch_type=ndimage_index,
+                                coercible=False)
```

Comment:
```suggestion
                                coercible=True)
```
required if the suggested change to `__ua_convert__` is made